airgun/entities/all_hosts.py

```diff
@@ -0,0 +1,62 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.utils import retry_navigation
+from airgun.views.all_hosts import (
+    AllHostsTableView,
+    BulkHostDeleteDialog,
+    HostDeleteDialog,
+)
+
+
+class AllHostsEntity(BaseEntity):
+    endpoint_path = '/new/hosts'
+
+    def search(self, host_name):
+        """Search for specific Host"""
+        view = self.navigate_to(self, 'All')
+        self.browser.plugin.ensure_page_safe(timeout='5s')
+        view.wait_displayed()
+        return view.search(host_name)
+
+    def delete(self, host_name):
+        """Delete host through table dropdown"""
+        view = self.navigate_to(self, 'All')
+        self.browser.plugin.ensure_page_safe(timeout='5s')
+        view.wait_displayed()
+        view.search(host_name)
+        view.table[0][2].widget.item_select('Delete')
+        delete_modal = HostDeleteDialog(self.browser)
+        if delete_modal.is_displayed:
+            delete_modal.confirm_delete.click()
+        view = self.navigate_to(self, 'All')
+        self.browser.plugin.ensure_page_safe(timeout='5s')
+        view.wait_displayed()
+        view.search(host_name)
+        return view.no_results
+
```

Comment:
This extra step of searching for host deleting after is unnecessary.

Reply:
@sambible Anyway we do a teardown/assertion at the test level for deletion so I believe @omkarkhatavkar is correct here.

Reply:
This is just to confirm that the host was actually deleted, and not errored out by satellite.